src/server/mod.rs

```diff
@@ -53,6 +53,12 @@ struct LiftService<T, B> {
     _pd: PhantomData<B>,
 }
 
+#[derive(Debug)]
+struct LiftServiceFuture<Fut, B> {
+    inner: Fut,
+    _pd: PhantomData<fn() -> B>,
```

Comment:
is there a specific reason you use the `fn() -> B` instead of just `B`?

Reply:
It is based on that the instance of `B` is returned from `Fut::poll` and `LiftServiceFuture` itself doesn't own its value.

it might not be necessary to have the type parameter `B` here...

Reply:
Yeah we need `B` here.